File: server/src/main/kotlin/pt/isel/g20/unicommunity/forumItem/model/ForumItem.java
package pt.isel.g20.unicommunity.forumItem.model;

import javax.persistence.*;

@Entity
public class ForumItem {

    private long boardId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String content;

    public ForumItem(){}

    public ForumItem(Long boardId, String name, String content){
        this.boardId = boardId;
        this.name=name;
        this.content = content;
    }

    public ForumItem(Long boardId, Long id, String name, String content){
        this.boardId = boardId;
        this.id = id;
        this.name = name;
        this.content = content;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getBoardId() {
        return boardId;
    }

    public void setBoardId(long boardId) {
        this.boardId = boardId;
    }
}

File: server/src/main/kotlin/pt/isel/g20/unicommunity/forumItem/model/ForumItem.kt
package pt.isel.g20.unicommunity.forumItem.model

import com.fasterxml.jackson.annotation.JsonFormat
import org.hibernate.annotations.CreationTimestamp
import pt.isel.g20.unicommunity.comment.model.Comment
import pt.isel.g20.unicommunity.forum.model.Forum
import pt.isel.g20.unicommunity.user.model.User
import java.util.*
import javax.persistence.*

@Entity
class ForumItem(
        @ManyToOne var forum: Forum,
        @ManyToOne @JoinColumn var author: User,
        @Column(nullable = false) var name: String,
        @Column(nullable = false) var content: String,
        @Column(nullable = false) var anonymousPost: Boolean
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @OneToMany(fetch = FetchType.LAZY)
    var comments: MutableList<Comment> = mutableListOf()

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd@HH:mm:ss.SSSZ", locale = "en_GB")
    @Column
    @CreationTimestamp
    var createdAt: Date? = null
}

